gph/python/ripser_interface.py

```diff
@@ -558,30 +558,27 @@ def ripser_parallel(X, maxdim=1, thresh=np.inf, coeff=2, metric="euclidean",
         if weights is not None:
             dm = _compute_weights(dm, weights, weight_params, n_points)
 
-        compute_enclosing_radius = False
+        apply_user_threshold = thresh != np.inf
         nonzero_in_diag = (dm.diagonal() != 0).any()
-        if not nonzero_in_diag:
+        if not nonzero_in_diag and not apply_user_threshold:
```

Comment:
@ulupo I prepare a suggestion in the logic for when to apply the threshold.
Let me know what you think of this and if the variables names make sense to you